src/com/providertrust/im981/Main.java

```diff
@@ -171,71 +191,144 @@ private static void reorderRecord(ArrayList<Tuple> tuples, PreparedStatement sel
+                                (tuple.getOrderId() == null ? -1 : tuple.getOrderId()),
                                 order,
                                 tuple.toString()));
-                        }
-                        break;
-                    case 1:
-                        assert tuples.contains(tuple);
-                        order = tuples.lastIndexOf(tuple) - 1; // 0 based
-                        if (orderId == null)
-                        {
-                            auditResult.updateInt(ORDER_COLUMN, order);
-                            auditResult.updateRow();
-                        }
-                        break;
-                    case 2:
-                        if(!tuples.remove(tuple))
-                        {
-                            log("REV-DELETE: !!DATA CORRUPTION!! Tuple doesn't exist -> " + tuple);
-                            // Bailing - data needs to be fixed first
-                            return;
-                        }
-
-                        if (!tuples.isEmpty())
-                        {
-                            for (int i = 0; i < tuples.size(); i++)
-                            {
-                                Tuple toCheck = tuples.get(i);
-                                //noinspection UnnecessaryUnboxing
-                                if (toCheck.getOrderId() == null || toCheck.getOrderId().intValue() != i)
-                                {
-                                    log(String.format("REV-DELETE: Re-order from %d to %d for %s",
-                                        (toCheck.getOrderId() == null ? -1 : toCheck.getOrderId()),
-                                        i,
-                                        toCheck.toString()));
-                                    toCheck.setOrderId(i);
-                                    updateOrder.setInt(1, i);
-                                    updateOrder.setInt(2, rev);
-                                    updateOrder.setLong(3, addressLinesId);
-                                    updateOrder.setString(4, line);
-                                    updateOrder.setInt(5, revType);
-                                    assert updateOrder.executeUpdate() == 1 : "Expected one row to be updated.";
-                                }
-                            }
-                        }
-
-                        break;
-                    default:
-                        throw new AssertionError("What's a " + revType + " revType");
+                        executeUpdateOrderId(updateOrder, tuple, order);
+                    }
+                    break;
+                case 2:
+                    if (!currentTuples.remove(tuple)) {
+                        log("REV-DELETE: !!DATA CORRUPTION!! Tuple doesn't exist -> " + tuple);
+                        // Bailing - data needs to be fixed first
+                        return true;
```

Comment:
Correct -- Since this is data corruption, we can not do anything thus just skip this one.